src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer`1.cs

```diff
@@ -47,7 +47,7 @@ internal bool TryParseValue(string value, [MaybeNullWhen(false)] out T result)
             if (optionalValue.HasValue)
             {
                 result = optionalValue.Value;
-                return result != null;
```

Comment:
This is no longer valid for case like Optional<bool?>()
Because if the input string is "null", and the parse result would be Optional<bool?>(null)

Reply:
Many options use 'nullable' case to represent the experiment, if we want to keep this, we might use a different value to represent the case, like 'default' or something that won't have collision in the future.

Reply:
Not sure what you mean. TryParseValue returns true if parsing was successful which is when _parseValue returns non-default Optional<T>. The result still may be null when true is returned.

Reply:
I mean this:
Suppose the type if bool? And the string needs to parse is `"null"`
_parseValue(value) would return `Optional<bool?>(null)` , `Optional` has a value, but the value is `null`
So here the parsing succeeds, but return `result != null` would give us `false`

Reply:
Oh, I missed that. Yeah, this should return `true`.